llama-index-core/llama_index/core/readers/file/base.py

```diff
@@ -20,6 +21,32 @@
 from tqdm import tqdm
 
 
+class BaseFilesystemReader(BaseReader, ABC):
+    @abstractmethod
+    def list_files(self, **kwargs) -> List[Path]:
+        """List files in the given filesystem."""
+
+    async def alist_files(self, **kwargs) -> List[Path]:
+        """List files in the given filesystem asynchronously."""
+        return self.list_files(**kwargs)
+
+    @abstractmethod
+    def get_file_info(self, input_file: Path, **kwargs) -> Dict:
```

Comment:
Am wondering if this should return a `FileInfo`/`FileMetadata` pydantic class which just contains some minimum required metadata fields to standardize this a bit 🤔